sensor/kubernetes/listener/listener_impl.go

```diff
@@ -82,15 +89,21 @@ func (k *listenerImpl) Stop(_ error) {
 	k.storeProvider.CleanupStores()
 }
 
-func clusterOperatorCRDExists(client client.Interface) (bool, error) {
-	resourceList, err := client.Kubernetes().Discovery().ServerResourcesForGroupVersion(clusterOperatorGroupVersion)
-	if err != nil {
-		return false, err
-	}
-	for _, apiResource := range resourceList.APIResources {
-		if apiResource.Name == clusterOperatorResourceName {
-			return true, nil
+func serverResourcesForGroup(client client.Interface, group string) (*metav1.APIResourceList, error) {
+	resourceList, err := client.Kubernetes().Discovery().ServerResourcesForGroupVersion(group)
+	return resourceList, err
+}
+
+// resourceExists returns true if resource exists in list.  Use with output from
+// `serverResourcesForGroup` to verify a resource exists prior to starting an
```

Comment:
if I have to call `serverResourcesForGroup` first, why not just make do something like this:

```
func resourceExists(client client.Interface, group, resource string) (bool, error) {
	resources, err := client.Kubernetes().Discovery().ServerResourcesForGroupVersion(group)
	if err != nil {
		return false, err
	}

	for _, apiResource := range list.APIResources {
		if apiResource.Name == resource {
			return true, nil
		}
	}

	return false, nil
```

Reply:
Because `client.Kubernetes().Discovery().ServerResourcesForGroupVersion()` does not cache the results, so that would mean for every invocation would repeat the same call to the k8s API.